projects/packages/device-detection/src/class-user-agent-info.php

```diff
@@ -277,6 +288,57 @@ public function get_platform() {
 		return $this->platform;
 	}
 
+	/**
+	 * Returns the platform for desktops
+	 *
+	 * @return string
+	 */
+	public function get_desktop_platform() {
+		$ua = $this->useragent;
```

Comment:
I've always wondered why half the methods in this class are static and access `$this->useragent` and such while the other half are static and directly access `$_SERVER['HTTP_USER_AGENT']`. Shouldn't it be one or the other everywhere?

This function seems particularly odd in that respect, since it checks if `$this->useragent` is set but then all the static functions it calls won't be using it.

Reply:
Fair point. In this case I just followed the existing style for the rest of the similar getters. I think some refactoring time on this class will be well spent as you're right that we are doing things in a not consistent way. Specifically looking for the `$this->useragent` will be okay as it is set in the constructor but overall I think it's confusing (and there is probably a reason for it to be this way) that we depend so much on a global var i.e. `$_SERVER['HTTP_USER_AGENT']`. Would be nice to see if we can just init things by passing a string and remove all these checks but also more importantly avoid setting global vars in the client code which can mess things up and create conflicts.

Reply:
We have been thinking the same, but although having a `$this->useragent` property indicates that it should be the source of the user agent, we have concerns about the issues that removing `$_SERVER['HTTP_USER_AGENT']` would cause to any existing users of the device detection functionality.